patches/minecraft/net/minecraft/item/ItemArmor.java.patch

```diff
@@ -1,5 +1,14 @@
 --- ../src-base/minecraft/net/minecraft/item/ItemArmor.java
 +++ ../src-work/minecraft/net/minecraft/item/ItemArmor.java
+@@ -187,7 +187,7 @@
+ 
+     public boolean func_82789_a(ItemStack p_82789_1_, ItemStack p_82789_2_)
+     {
+-        return this.field_77878_bZ.func_151685_b() == p_82789_2_.func_77973_b() ? true : super.func_82789_a(p_82789_1_, p_82789_2_);
++        return !this.field_77878_bZ.getRepairItemStack().func_190926_b() ? true : super.func_82789_a(p_82789_1_, p_82789_2_);
```

Comment:
You removed the `OreDictionary.itemMatches` check here, that's not right.
The check you have here will accept any material as long as a repair material exists.
It needs to compare the `p_82789_2_` parameter with the repair material using `OreDictionary.itemMatches`.